app/models/testcase.js

```diff
@@ -240,7 +240,9 @@ TestCaseSchema.static({
  * Register
  */
 TestCaseSchema.plugin( QueryPlugin ); //install QueryPlugin
+
 var Testcase = mongoose.model('Testcase', TestCaseSchema);
-Testcase.ensureIndexes(function (err) {
-  if (err) return handleError(err);
-});
+// TODO figure out how to synchronously ensureIndexes so we do not close db connection before the process is completed
```

Comment:
its still better to be asynchronous. Solution could be that we emit some common event bus message that indexes is ensured before allow clients to be connected to API's...  -> "kind of boot sequence"...